### PR TITLE
Jenkins UI p1

### PR DESCRIPTION
**Pull request by sadmananik**
This PR Contains Black Duck Installation Path, Coverity Installation Path, Synopsys Bridge Version (Optional), Synopsys Bridge Installation Path (Optional) fields.

### CODE REVIEW

File: src/main/resources/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig/config.jelly [hunk 1]
```diff
@@ -31,5 +37,11 @@
         <f:entry field="synopsysBridgeDownloadUrl" title="Synopsys Bridge Download URL (Optional)">
             <f:textbox/>
         </f:entry>
+        <f:entry field="synopsysBridgeVersion" title="Synopsys Bridge Version (Optional)">
+            <f:textbox/>
+        </f:entry>
+        <f:entry field="synopsysInstallationPath" title="Synopsys Bridge Installation Path (Optional)">
```
Comment (maksudur-rahman-maruf) [inline src/main/resources/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig/config.jelly:43]: I think if we name it as synopsysBridgeInstallationPath, it makes more sense.

File: src/main/java/com/synopsys/integration/jenkins/scan/service/ScanCommandsFactory.java [hunk 2]
```diff
@@ -104,6 +110,14 @@ private static Map<String, Object> getGlobalConfigurationValues() {
                 globalParameters.put(ApplicationConstants.BRIDGE_DOWNLOAD_URL, config.getSynopsysBridgeDownloadUrl());
             }
 
+            if (!Utility.isStringNullOrBlank(config.getSynopsysInstallationPath())) {
```
Comment (maksudur-rahman-maruf) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/ScanCommandsFactory.java:113]: same

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 3]
```diff
@@ -141,10 +180,22 @@ public String getCoverityConnectUserPassword() {
...
+
     public String getSynopsysBridgeDownloadUrl() {
         return synopsysBridgeDownloadUrl;
     }
 
+    public String getSynopsysBridgeVersion() {
+        return synopsysBridgeVersion;
+    }
+
+    public String getSynopsysInstallationPath() {
```
Comment (maksudur-rahman-maruf) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:195]: same

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 4]
```diff
@@ -39,10 +42,19 @@ public class ScannerGlobalConfig extends GlobalConfiguration implements Serializ
...
+
     @HelpMarkdown("Provide the Synopsys Bridge artifactory URL from where synopsys-bridge zip file can be downloaded. " +
         "Note: You must need to provide the full download url which includes the zip file path.")
     private String synopsysBridgeDownloadUrl;
 
+    @HelpMarkdown("Provide the specific Synopsys bridge version you want to download from artifactory.")
+    private String synopsysBridgeVersion;
+
+    @HelpMarkdown("Provide the Synopsys bridge installation path where it will be installed.")
+    private String synopsysInstallationPath;
```
Comment (maksudur-rahman-maruf) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:56]: same

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 5]
```diff
@@ -105,6 +129,18 @@ public void setSynopsysBridgeDownloadUrl(String synopsysBridgeDownloadUrl) {
...
     }
 
+    @DataBoundSetter
+    public void setSynopsysBridgeVersion(String synopsysBridgeVersion) {
+        this.synopsysBridgeVersion = synopsysBridgeVersion;
+        save();
+    }
+
+    @DataBoundSetter
+    public void setSynopsysInstallationPath(String synopsysInstallationPath) {
```
Comment (maksudur-rahman-maruf) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:139]: same

**Review by jahid1209 — CHANGES_REQUESTED** [review]

**Review by zaman-akib — APPROVED** [review]

**Review by jahid1209 — APPROVED** [review]

**Pull request merged**